### DISCUSSION TITLE
[Lua_Ls] Define globals that are called with REQUIRE automatically

### DISCUSSION THREAD

**Discussion by rudfol** [LazyVim/LazyVim#609] [Q&A]
Hello there,

Simply I could define a global 'lpeg' for example but after a while, with every new 'require' of a library, this would get
tedious. I'm just wondering if there are ways to recognise a library has been called with 'require' then those globals as a bit of
an automatic process?

![1 boom](https://user-images.githubusercontent.com/6250738/232554833-305304e5-6489-4391-a5f2-5678297b01bb.PNG)

What I have in my config so far.

![1 boom2](https://user-images.githubusercontent.com/6250738/232576765-5c693d53-4005-40ef-97fc-8d77c8340dbb.PNG)

**Comment by folke** [comment]
Sorry, but this has nothing to do with LazyVim. I don't even understand what you're trying to achieve here. Check the lua_ls docs.

**Comment by rudfol** [comment]
Sorry, I meant I'm wondering whether it's possible to define all globals in a lua file. Depending on how many imported libraries there are in a lua file. It might be tedious to do it by hand one by one each time I use more libraries. kudos
